client/container_wait_test.go

```diff
@@ -1,12 +1,58 @@
 package client
 
 import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
 	"log"
+	"net/http"
+	"testing"
 	"time"
 
+	"github.com/docker/engine-api/client/transport"
+	"github.com/docker/engine-api/types"
+
 	"golang.org/x/net/context"
 )
 
+func TestContainerWaitError(t *testing.T) {
+	client := &Client{
+		transport: transport.NewMockClient(nil, transport.ErrorMock(http.StatusInternalServerError, "Server error")),
+	}
+	code, err := client.ContainerWait(context.Background(), "nothing")
+	if err == nil || err.Error() != "Error response from daemon: Server error" {
+		t.Fatalf("expected a Server Error, got %v", err)
+	}
+	if code != -1 {
+		t.Fatalf("expected a status code equal to '1', got %d", code)
```

Comment:
In case of errors, `ContainerWait` returns `-1` as *statusCode* (https://github.com/docker/engine-api/blob/master/client/container_wait.go#L16), though it was worth checking :sweat_smile: